src/model.ts

```diff
@@ -510,7 +510,8 @@ export class SpraypaintBase {
       Object.keys(attrs).forEach(k => {
         let self = this as any
         let changes = this.changes() as any
-        if (self[k] !== attrs[k] && !changes[k]) {
+        let attrDef = this.klass.attributeList[k]
+        if (attrDef.dirtyChecker(self[k], attrs[k]) && !changes[k]) {
```

Comment:
Not sure about this one but it felt right.
It also makes me think the name is not good enough since it does not really check if the attribute is dirty.